src/transformers/dynamic_module_utils.py

```diff
@@ -149,6 +149,8 @@ def get_imports(filename: Union[str, os.PathLike]) -> List[str]:
 
     # filter out try/except block so in custom code we can have try/except imports
     content = re.sub(r"\s*try\s*:\s*.*?\s*except\s*.*?:", "", content, flags=re.MULTILINE | re.DOTALL)
+    # filter out imports under is_flash_attn_2_available block for avoid import issues in cpu only environment
+    content = re.sub(r"if is_flash_attn_*available\(\):\s*(from flash_attn\s*.*\s*)+", "", content, flags=re.MULTILINE)
```

Comment:
I believe, we could improve the regex to `r"if is_flash_attn_[0-9]+_available\(\):\s*(from flash_attn\s*.*\s*)+"`
I was trying to use this change locally for OrionStarAI/Orion-14B-Chat and the flash_attn imports were not removed.
Most probably, the similar stands for the other models mentioned in the ticket as the imports string are the same.

Reply:
sorry, but I lost any motivation to finishing this PR after 3 months o silence... I'm not good in regular expressions, so I have no time to explore all models on hub to find all possible patterns...

I'll apply suggestion from @CuriousPanCake, but do not promise to look anything beyond of that if you do not have specific suggestions.